subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
+{
+	struct net_if *iface = net_if_get_default();
+
+	if (!iface) {
+		NET_ERR("Cannot bind to %s",
+			net_sprint_ipv6_addr(&addr6->sin6_addr));
+
+		return -EADDRNOTAVAIL;
```

Comment:
The interface checking needs to be done differently, perhaps something like this
```
if (!net_context_get_iface(context)) {
       net_context_set_iface(context, net_if_get_default());

       if (!net_context_get_iface(context)) {
                return -ENOENT;
       }
}
```
although I am not sure if we really need to check if the default interface returned NULL, probably NET_ASSERT() would be sufficient here.